crates/resolver/src/hosts.rs

```diff
@@ -158,12 +158,21 @@ impl Hosts {
         // if line only include `addr` without `host` will be ignored,
         // the src will be parsed to map in the form `Name -> LookUp`.
 
-        for line in BufReader::new(src).lines() {
-            // Remove comments from the line
+        for (line_index, line) in BufReader::new(src).lines().enumerate() {
             let line = line?;
+
+            // Remove byte-order mark if present
+            let line = if line_index == 0 && line.starts_with('\u{feff}') {
+                // BOM is 3 bytes
+                &line[3..]
+            } else {
+                &line
+            };
```

Comment:
I don't think this solution makes sense. `\u{feff}` is the big-endian UTF-16 encoding of the BOM, whereas `BufReader::lines()` implicitly relies on an UTF-8 encoding of the file. So IMO if we want to correctly handle UTF-16-encoded files (with or without a BOM), this is not the right approach.

Reply:
Here, we're checking for a `char` at the beginning of a string, so this is happening after UTF-8 encoding, and we're not looking at the underlying bytes. U+FEFF is the correct codepoint for the BOM. With that said, I recall Windows Notepad defaulting to UTF-16 encoding for files with non-ASCII characters in the past, so it might be worth peeking at the first few bytes to see if the encoding can be autodetected from the presence of a BOM.

Reply:
I'm not sure what you mean by "after UTF-8 encoding". If the file on disk is UTF-16 encoded, then `BufReader::lines()` has already interpreted as UTF-8 which seems problematic in this case?

Reply:
Sorry, should have said "after UTF-8 decoding". My point is that `'\u{feff}'` is not the big-endian UTF-16 encoding, because it is the decoded character itself. Nothing in this PR uses or assumes UTF-16 so far, but it might be worthwhile to support that in addition.

Reply:
Ahh, okay, that's fair. So let's merge this and leave UTF-16 support for when we need it. I'm guessing non-ASCII text in the hosts file is going to be pretty rare.